File: idrougge-kotlin/day1_2.kt
/* Advent of code 2018, day 1, part 2 in Kotlin 1.3 */
import java.io.File

fun main() {
	val fn = "day1.txt"
	val changes = File(fn).readLines().map(String::toInt).toMutableList()
	val seen = mutableSetOf<Int>()

	val seq = generateSequence(0) {
		val change = changes.removeAt(0)
		changes.add(change)
		it + change
	}
	println(seq.find{ !seen.add(it) })
}

File: idrougge-swift/day1_2.swift
/* Advent of code 2018, day 1, part 2 in Swift 4 */

import Foundation

let changes = try! String(contentsOfFile: "day1.txt")
    .components(separatedBy: .newlines)
    .flatMap(Int.init)

var freq = 0
var freqs: Set<Int> = []

while true {
    if !changes.drop(while: { change in
        freq += change
        return freqs.insert(freq).inserted
        }).isEmpty {
        break
    }
}

print(freq)
